src/systems/systems/src/air_conditioning/acs_controller.rs

```diff
@@ -0,0 +1,1808 @@
+        self.should_open_fcv = self.should_open_fcv_determination(engines);
+    }
+
+    fn pack_start_condition_determination(&mut self, pack_flow_valve: &[PackFlowValve; 2]) -> bool {
+        // Returns true when one of the packs is in start condition
+        pack_flow_valve
+            .iter()
+            .any(|fcv| fcv.fcv_timer() <= Duration::from_secs_f64(Self::PACK_START_TIME))
+    }
+
+    fn flow_demand_determination(&self, aircraft_state: &AcStateManager) -> Ratio {
+        let mut intermediate_flow =
+            Ratio::new::<percent>((self.flow_selector_position as u32) as f64);
+        // TODO: Add "insufficient performance" based on Pack Mixer Temperature Demand
+        if self.pack_in_start_condition {
+            intermediate_flow =
+                intermediate_flow.max(Ratio::new::<percent>(Self::PACK_START_FLOW_LIMIT));
+        }
+        if self.apu_bleed_on {
+            intermediate_flow =
+                intermediate_flow.max(Ratio::new::<percent>(Self::APU_SUPPLY_FLOW_LIMIT));
+        }
+        if self.single_pack_operation {
+            intermediate_flow =
+                intermediate_flow.max(Ratio::new::<percent>(Self::ONE_PACK_FLOW_LIMIT));
+        }
+        if matches!(
+            aircraft_state,
+            AcStateManager::BeginTakeOff(_)
+                | AcStateManager::EndTakeOff(_)
+                | AcStateManager::BeginLanding(_)
+                | AcStateManager::EndLanding(_)
+        ) {
+            intermediate_flow =
+                intermediate_flow.min(Ratio::new::<percent>(Self::FLOW_REDUCTION_LIMIT));
+        }
+        intermediate_flow.max(Ratio::new::<percent>(Self::BACKFLOW_LIMIT))
+    }
+
+    // This calculates the flow based on the demand, when the packs are modelled this needs to be changed
+    // so the demand actuates the valve, and then the flow is calculated based on that
+    fn absolute_flow_calculation(&self, pressurization: &impl CabinAltitude) -> MassRate {
+        let absolute_flow = self.flow_demand.get::<ratio>()
+            * (Self::FLOW_CONSTANT_XCAB * pressurization.cabin_altitude().get::<foot>()
+                + Self::FLOW_CONSTANT_C);
+        MassRate::new::<kilogram_per_second>(absolute_flow)
+    }
+
+    fn fcv_open_allowed_determination(&mut self, acs_overhead: &AirConditioningSystemOverhead) {
+        // Flow Control Valve 1
+        self.fcv_1_open_allowed = acs_overhead.pack_1_pb_is_on()
+            && !self.engine_1_in_start_mode
+            && (!self.engine_2_in_start_mode || !self.crossbleed_is_on)
+            && !self.engine_1_on_fire
+            && !self.ditching_is_on;
+        // && ! pack 1 overheat
+        // Flow Control Valve 2
+        self.fcv_2_open_allowed = acs_overhead.pack_2_pb_is_on()
+            && !self.engine_2_in_start_mode
+            && (!self.engine_1_in_start_mode || !self.crossbleed_is_on)
+            && !self.engine_2_on_fire
+            && !self.ditching_is_on;
+        // && ! pack 2 overheat
```

Comment:
**thought**: This is declared in the `systems` crate and thus should also work for other aircraft types (e.g. A380 with 4 engines). Is there a way to abstract this a bit?

Reply:
I've changed this slightly to read "right" and "left" engines instead of 1 and 2. The A380 also has 2 packs, so we should be able to use the same code just by passing the state of the corresponding engines (i.e. `pneumatic.left_engine_state() == EngineState::Starting` means engine 1 on the A320, and will mean any of engines 1 or 2 in the A380).

Having said this, I don't know enough about the A380 yet to promise this won't need any tweaks.